src/Node/ForNode.php

```diff
@@ -101,7 +101,7 @@ public function compile(Compiler $compiler): void
         $compiler->write("\$_parent = \$context['_parent'];\n");
 
         // remove some "private" loop variables (needed for nested loops)
-        $compiler->write('unset($context[\'_seq\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']');
+        $compiler->write('unset($context[\'_parent\'], $context[\'_seq\'], $context[\''.$this->getNode('key_target')->getAttribute('name').'\'], $context[\''.$this->getNode('value_target')->getAttribute('name').'\'], $context[\'_parent\']');
```

Comment:
This is done at the end of the line :)

Reply:
Thanks for spotting this. I didn't realize it and was too quick with the PR.